apps/frontend/src/auth/auth.utils.ts

```diff
@@ -50,3 +55,19 @@ export const loginFormValidator = (form: LoginForm) => {
 
     return errors;
 };
+
+export const registerFormValidator = (form: RegisterForm) => {
+    const errors: RegisterValidationErrors = {};
+    const emailRegex = /^[\w-]{1,30}@([\w-]+\.)+[\w-]{2,4}$/g;
+    if (!emailRegex.test(form.email)) {
+        errors.email = 'Email is not in a valid format';
+    }
+    if (form.password.length < 8) {
+        errors.password = 'Password must be at least 8 characters';
+    }
+    if (form.username.length < 3) {
+        errors.username = 'Username must be at least 3 characters';
+    }
+
+    return errors;
+};
```

Comment:
It is correct :). You asked how to do it in other way... the other solution is to handle errors as soon as the input changes and if we have any error set to true then we fail in validation method.